### ISSUE TITLE
How to search instances by filters like date, status ?

### ISSUE THREAD

**Issue by hsnsalhi** [Azure/durabletask#674]
Hello, 
I want to query orchestration instances with some filters like creation time, completion time, orchestration status.
This has been implemented in the mssql provider but I want to know if this could be at the DTF level by providing query interfaces to be implemented in each persistence provider.

**Comment by cgillum** [comment]
Indeed, this is also supported by the [AzureStorageOrchestrationService](https://github.com/Azure/durabletask/blob/6c1fda6431b1f27cf1e2c5359a0dca62b9202d31/src/DurableTask.AzureStorage/AzureStorageOrchestrationService.cs#L1670-L1698), but as you said it's not exposed to TaskHubClient, nor to `IOrchestrationServiceClient`.  It can be added, but it would likely require introducing a new public interface with a common set of query filter options.
